### ISSUE TITLE
automatically accept registrations from invitations

### ISSUE THREAD

**Issue by atdqm** [humhub/humhub#5902]
### Is your feature request related to a problem? Please describe.
no

### Describe the solution you'd like
When using HumHub for a trusted community, allow members to invite others who will be automatically accepted while non-invited registrations must be approved.

If the option "Require group admin approval after registration" is checked, add another option "Automatically approve users invited by registered members".

### Describe alternatives you've considered
For security, automatic acceptances can be restricted to invitations from authorized groups only

**Comment by funkycram** [comment]
@luke- I can develop this request if you agree.
I would will an option in `/admin/authentication`, just after "Require group admin approval after registration": "Automatically approve users invited by registered members"

**Comment by atdqm** [comment]
Idea from another user : 
https://community.humhub.com/s/general-discussion/?contentId=261084&commentId=43393
that looks interesting, but i was hoping to only allow members of certain groups to even see the ‘invite by email’ option..

**Comment by dantefromhell** [comment]
For one upcoming platform idea it would be helpful to be able to whitelist domains so for emails within that domain admin approval is not required, but for others.

In summary, this might be an asking for flexible exception rules to the admin approval setting?

**Comment by atdqm** [comment]
Yes the idea is to find a way to approve automatically but not everybody.

Our community is a "public" community, so they have  emails on gmail.com, hotmail.com, outlook.com and so on...

the whitelist is not enough.

**Comment by ArchBlood** [comment]
> For one upcoming platform idea it would be helpful to be able to whitelist domains so for emails within that domain admin approval is not required, but for others.
> 
> In summary, this might be an asking for flexible exception rules to the admin approval setting?

What you're suggesting is already possible with HumHub Professional Edition;

https://www.humhub.com/en/marketplace/email-whitelist/

**Comment by ArchBlood** [comment]
> yes but if you have 2000 - 3000 emails to put in the whitelist, the field won't accept them all.  
> 
> the whitelist is good for domains (*@mycompany.com)  but not for emails one by one...

You shouldn't need to enter 2000-3000 email addresses, all you should need is a single domain, `gmail.com`, `hotmail.com` and etc, then all emails should be whitelisted under that said domain.

**Comment by atdqm** [comment]
but I don't want to accept all the user of gmail, only the ones of my community who have a gmail email...

**Comment by luke-** [comment]
I like the automatic approval of invited users. Should also be quite easy to implement. 
Maybe we should then also save who was invited by whom.

**Comment by atdqm** [comment]
> Maybe we should then also save who was invited by whom.

It already exists : the administrators have a "list pending registration" in the bottom of the users page 
![image](https://user-images.githubusercontent.com/112695767/200283303-ff0667f2-340e-4be7-94ab-3f33b5d47314.png) 
with a "invited by" column
![image](https://user-images.githubusercontent.com/112695767/200283434-c1ccaf99-09e6-4ad2-b127-a07dcd789cd8.png)
